src/tests-bns/event-server-tests.ts

```diff
@@ -1069,14 +1069,14 @@ describe('BNS event server tests', () => {
     expect(configState.bns_subdomains_imported).toBe(false)
 
     await new Promise(resolve => {
-      db.eventEmitter.on('bnsImportUpdate', (bnsNamesOnchainImported, bnsSubdomainsImported) => {
-        if (bnsNamesOnchainImported && bnsSubdomainsImported) {
-          expect(bnsNamesOnchainImported).toBe(true)
-          expect(bnsSubdomainsImported).toBe(true);
+      db.eventEmitter.on('configStateUpdate', (configState) => {
+        if (configState.bns_names_onchain_imported && configState.bns_subdomains_imported) {
+          expect(configState.bns_names_onchain_imported).toBe(true)
+          expect(configState.bns_subdomains_imported).toBe(true);
           resolve(undefined);
         }
```

Comment:
You'll just need an `else` with a reject so the test actually fails if this is the case

Reply:
This event can be emitted when one or both of these conditions are false. In fact, this is expected because the names will be imported first, and then the subdomains. So, the first time that this check runs, subdomains will be false. We don't want to reject because we are exepcting subdomains to be imported next. This test will fail with a timeout if we never hit resolve.